xcube/util/cmaps.py

```diff
@@ -444,48 +465,55 @@ def _register_ocm_cmaps(self):
 def parse_cm_code(cm_code: str) -> tuple[str, Optional[Colormap]]:
     # Note, if we get performance issues here, we should
     # cache cm_code -> colormap
+    values: Optional[list[Union[int, float]]] = None
     try:
         user_color_map: dict[str, Any] = json.loads(cm_code)
         cm_name = user_color_map["name"]
         cm_items = user_color_map["colors"]
         cm_type = user_color_map.get("type", "node")
         cm_base_name, _, _ = parse_cm_name(cm_name)
-        if cm_type == "key" or cm_type == "bound":
-            if cm_type == "key":
-                bounds: list[int] = []
-                colors: list[str] = []
-                n = len(cm_items)
-                for i, (value, color) in enumerate(cm_items):
-                    index = int(value)
-                    colors.append(color)
-                    bounds.append(index)
-                    if i == n - 1 or index + 1 != int(cm_items[i + 1][0]):
-                        # insert transparent region
-                        bounds.append(index + 1)
-                        colors.append("#00000000")
-            else:  # cm_type == "bound"
-                bounds: list[Union[int, float]] = list(map(lambda c: c[0], cm_items))
-                colors: list[str] = list(map(lambda c: c[1], cm_items[:-1]))
-            return cm_name, Colormap(
-                cm_base_name,
-                cat_name=CUSTOM_CATEGORY.name,
-                cmap=matplotlib.colors.ListedColormap(colors),
-                bounds=bounds,
-            )
-        else:
+        n = len(cm_items)
+        if cm_type == "key":
+            values: list[int] = []
+            colors: list[str] = []
+            for i, (value, color) in enumerate(cm_items):
+                key = int(value)
+                values.append(key)
+                colors.append(color)
+                if i == n - 1:
+                    # the last key's next boundary is key+1
+                    values.append(key + 1)
+                elif key + 1 != int(cm_items[i + 1][0]):
+                    # insert transparent region from key+1 to next key
+                    values.append(key + 1)
+                    colors.append("#00000000")
+            cmap = matplotlib.colors.ListedColormap(colors, name=cm_base_name)
```

Comment:
This does not work properly. 

See example below. The values at the coast are all above 6, as presented in the continuous colormap on the left. However, the categorical colormap on the right finds values for category 4 and category 2. 

![Screenshot from 2024-07-18 09-14-47](https://github.com/user-attachments/assets/c706860f-d263-467f-b3a6-35dac2d6bac2)

I think also here, we will not get around using `matplotlib.colors.BoundaryNorm` (see PR #1019)

Reply:
if `cm_type == "key"`, I indeed use `matplotlib.colors.BoundaryNorm`.